tests/test_lock_queue.py

```diff
@@ -0,0 +1,117 @@
+import builtins
+import copy
+import os
+import sys
+import unittest
+import unittest.mock as mock
+
+import lock
+import lock_queue
+
+## Instead of saving the process queue to disk the mocks will save it here.
+## It needs to be global because it is shared between threads.
+saved_queue = []
+
+def mock_pickle_dump_fn(*args):
+   global saved_queue
+   saved_queue = copy.deepcopy(args[0])
+
+def mock_pickle_load_fn(*args):
+   global saved_queue
+   return copy.deepcopy(saved_queue)
+
+
+class Test_LockQueue(unittest.TestCase):
+    def setUp(self):
+        # Re-initialize queue to empty for each test
+        global saved_queue
+        saved_queue = []
+
+    def get_lock_name(self):
+        return "bacon"
+
+    def tearDown(self):
+        # Make sure the test didn't leave anything behind on the filesystem
+        mem_file = f"/tmp/mem-{self.get_lock_name()}"
+        #self.assertFalse(os.path.exists(mem_file))
+
+
+    @mock.patch('lock_queue.pickle.load', side_effect=mock_pickle_load_fn)
+    @mock.patch('lock_queue.pickle.dump', side_effect=mock_pickle_dump_fn)
+    @mock.patch('lock_queue.os.path.exists')
+    @mock.patch('lock.Lock', autospec=False)
+    def test_will_init_queue(self, lock, os_path_exists, pdump, pload):
+        # Add a fake value to queue
+        global saved_queue
+        saved_queue = [911]
+
+        # Pretend the queue backing file does not exist
+        os_path_exists.return_value = False
+
+        lq = lock_queue.LockQueue(self.get_lock_name())
+
+        # LockQueue constructor should have re-initialized the queue to empty
+        self.assertEqual(saved_queue, [])
+
+
+    @mock.patch('lock_queue.pickle.load', side_effect=mock_pickle_load_fn)
+    @mock.patch('lock_queue.pickle.dump', side_effect=mock_pickle_dump_fn)
+    @mock.patch('lock_queue.os.getpid')
+    @mock.patch('lock_queue.get_process_start_time')
+    @mock.patch('lock.Lock', autospec=False)
+    def test_push_to_queue_3x(self, lock, start_time, getpid, pdump, pload):
+        global saved_queue
+
+        lq = lock_queue.LockQueue(self.get_lock_name())
+        # Push to queue 3 times using these PID and Start Time combinations
+        test_pids = [993, 996, 997]
```

Comment:
might be better if the pids were not in increasing order here. I'm sure the code functions correctly but it would make it clear and guard against any future change which accidentally sorted the queue for instance.